lib/Common/Memory/Recycler.h

```diff
@@ -152,25 +150,6 @@ struct InfoBitsWrapper{};
 #define RecyclerNewWithBarrierFinalizedClientTracked(recycler,T,...) static_cast<T *>(static_cast<FinalizableObject *>(AllocatorNewBase(Recycler, recycler, AllocFinalizedClientTrackedWithBarrierInlined, T, __VA_ARGS__)))
 #endif
 
-
-#if defined(RECYCLER_WRITE_BARRIER_ALLOC) && GLOBAL_ENABLE_WRITE_BARRIER
-#define RecyclerNew                         RecyclerNewWithBarrier
-#define RecyclerNewPlus                     RecyclerNewWithBarrierPlus
-#define RecyclerNewPlusZ                    RecyclerNewWithBarrierPlusZ
-#define RecyclerNewZ                        RecyclerNewWithBarrierZ
-#define RecyclerNewStruct                   RecyclerNewWithBarrierStruct
-#define RecyclerNewStructZ                  RecyclerNewWithBarrierStructZ
-#define RecyclerNewStructPlus               RecyclerNewWithBarrierStructPlus
-#define RecyclerNewArray                    RecyclerNewWithBarrierArray
-#define RecyclerNewArrayZ                   RecyclerNewWithBarrierArrayZ
-#define RecyclerNewFinalized                RecyclerNewWithBarrierFinalized
-#define RecyclerNewFinalizedPlus            RecyclerNewWithBarrierFinalizedPlus
-#define RecyclerNewTracked                  RecyclerNewWithBarrierTracked
-#define RecyclerNewEnumClass                RecyclerNewWithBarrierEnumClass
-#define RecyclerNewWithInfoBits             RecyclerNewWithBarrierWithInfoBits
-#define RecyclerNewFinalizedClientTracked   RecyclerNewWithBarrierFinalizedClientTracked
-#endif
```

Comment:
I think we should make the change, currently we need the assumption that all recycler allocation are made through RecyclerNew* macros, to get the plugin work. what if there's recycler->Alloc() call? (I don't think there is one for now though), I remember there's RecyclerFastAllocator might be close to the pattern "recycler->Alloc", let me check, may be we need to do something special there

---
In reply to: [89900815](https://github.com/Microsoft/ChakraCore/pull/2105#discussion_r89900815) [](ancestors = 89900815)